commcare_connect/opportunity/tables.py

```diff
@@ -549,3 +553,285 @@ def date_with_time_popup(table, date):
+            text = "Active"
+        elif value == 1:
+            badge_class = "badge badge-sm bg-orange-600/20 text-orange-600"
+            text = "Ended"
+        else:
+            badge_class = "badge badge-sm bg-slate-100 text-slate-400"
+            text = "Inactive"
+
+        return format_html(
+            '<div class="flex justify-start text-sm font-normal truncate text-brand-deep-purple overflow-clip overflow-ellipsis">'
+            '  <span class="{}">{}</span>'
+            "</div>",
+            badge_class,
+            text,
+        )
+
+    def _render_div(self, value, extra_classes=""):
+        base_classes = "flex text-sm font-normal truncate text-brand-deep-purple " "overflow-clip overflow-ellipsis"
+        all_classes = f"{base_classes} {extra_classes}".strip()
+        return format_html('<div class="{}">{}</div>', all_classes, value)
+
+    def render_opportunity(self, value):
+        return self._render_div(value, extra_classes="justify-start")
+
+    def render_program(self, value):
+        return self._render_div(value if value else "--", extra_classes="justify-start")
+
+    def render_start_date(self, value):
+        return self._render_div(value, extra_classes="justify-center")
+
+    def render_end_date(self, value):
+        return self._render_div(value, extra_classes="justify-center")
+
+
+class OpportunityTable(BaseOpportunityList):
+    pending_invites = tables.Column()
+    inactive_workers = tables.Column()
+    pending_approvals = tables.Column()
+    payments_due = tables.Column()
+    actions = tables.Column(empty_values=(), orderable=False, verbose_name="")
+
+    class Meta(BaseOpportunityList.Meta):
+        sequence = BaseOpportunityList.Meta.sequence + (
+            "pending_invites",
+            "inactive_workers",
+            "pending_approvals",
+            "payments_due",
+            "actions",
+        )
+
+    def render_pending_invites(self, value):
+        return self._render_div(value, extra_classes=self.stats_style)
+
+    def render_inactive_workers(self, value):
+        return self._render_div(value, extra_classes=self.stats_style)
+
+    def render_pending_approvals(self, value):
+        return self._render_div(value, extra_classes=self.stats_style)
+
+    def render_payments_due(self, value):
+        if value is None:
+            value = 0
+        return self._render_div(f"${value}", extra_classes=self.stats_style)
+
+    def render_actions(self, record):
+        actions = [
+            {
+                "title": "View Opportunity",
+                "url": reverse("opportunity:detail", args=[record.organization.slug, record.id]),
+            },
+            {
+                "title": "View Workers",
+                "url": reverse("opportunity:detail", args=[record.organization.slug, record.id]),
+                # "url": reverse("opportunity:tw_worker_list", args=[record.organization.slug, record.id]),
```

Comment:
do we need the commented out code?

Reply:
Yeah, this is updated in the followup PR [here](https://github.com/dimagi/commcare-connect/pull/573/files#diff-04b178fa6d9c4a6615700a161f3de90ed4ba7650547e3590dafd5eb4e17019a2R735)